src/ledger.ts

```diff
@@ -252,16 +233,14 @@ export class LedgerInteraction extends DirectKeystoreInteraction {
    *
    * See the [Ledger API]{@link https://github.com/LedgerHQ/ledgerjs} for general information or the [bitcoin app API]{@link https://github.com/LedgerHQ/ledgerjs/tree/master/packages/hw-app-btc} for examples of API calls.
    *
-   * @param {function} callback -- accepts two parameters, `app` and `transport`, which are the Ledger APIs for the bitcoin app and the transport layer, respectively.
-   * @returns {Promise} does the work of setting up an app instance (and transport connection)
    * @example
    * async run() {
    *   return await this.withApp(async (app, transport) => {
    *     return app.doSomething(); // Not a real Ledger bitcoin app API call
    *   });
    * }
    */
-  withApp(callback) {
+  withApp(callback: (app: any, transport: any) => any) {
```

Comment:
app is definitely available from ledger [here](https://github.com/unchained-capital/unchained-wallets/blob/master/src/ledger.ts#L57). `AppClient`.

Reply:
Actually, `AppClient` doesn't work. I changed this an there were 5 lines erroring on reference of methods on the class type. For example: `Property 'getWalletPublicKey' does not exist on type 'AppClient'.`

Reply:
That's strange. I wonder if it's because of the old app version or something.